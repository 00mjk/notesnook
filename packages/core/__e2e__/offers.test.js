import hosts from "../utils/constants";
import Offers from "../api/offers";

test("get offer code", async () => {
  const offers = new Offers();
  hosts.SUBSCRIPTIONS_HOST = "https://subscriptions.streetwriters.co";
  expect(await offers.getCode("TESTOFFER", "android")).toMatchSnapshot(
    "offer-code"
  );
});

test("get invalid offer code", async () => {
  const offers = new Offers();
  hosts.SUBSCRIPTIONS_HOST = "https://subscriptions.streetwriters.co";
  await expect(() => offers.getCode("INVALIDOFFER", "android")).rejects.toThrow(
    /Not found/i
  );
});
