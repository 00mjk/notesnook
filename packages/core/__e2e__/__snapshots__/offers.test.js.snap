// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get offer code: offer-code 1`] = `"123"`;
